myapp/spec/system/tasks_searching_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe Task, type: :model do
+  describe "search_by_name_and_status" do
+    # Create a user
+    let(:user) { create(:user) }
+    # Create tasks with different creation and dealine timestamps for testing sorting
+    let!(:task1) { create(:task, name: "Task 1", status: "Not Started", user: user) }
+    let!(:task2) { create(:task, name: "Task 2", status: "In Progress", user: user) }
+    let!(:task3) { create(:task, name: "Another Task", status: "Done", user: user) }
+
+    it "returns tasks that match the search by name and status" do
```

Comment:
The system spec test should be written to simulate user actions.
Enter text in the name field, select a status, click the search button, and check the expected UI.
I think this test should be written in the model spec.

Reply:
Sure, I move this test case to model spec and add new system spec as for checking the the search function
<img width="424" alt="image" src="https://github.com/Fablic/training/assets/88306119/d336ce56-be68-4a49-b9a8-31c5bde9596a">